inc/thorvg.h

```diff
@@ -1544,16 +1544,42 @@ class TVG_API Text final : public Paint
      *
      * @note Experimental API
      *
-     * @see Text::unload(const std::string& path)
+     * @see Text::unload(const std::string& fontId)
      */
     static Result load(const std::string& path) noexcept;
 
+    /**
+     * @brief Loads a scalable font data (ttf) from a memory block of a given size.
+     *
+     * ThorVG efficiently caches the loaded font data using the specified @p name as a key when @p copy
+     * is set to @c false. This means that loading the same fonts again will not result in duplicate
+     * operations for the shared @p data. Instead, ThorVG will reuse the previously loaded font data.
+     *
+     * @param[in] name The name under which the font will be stored and accessible.
+     * @param[in] data A pointer to a memory location where the content of the ttf data is stored.
+     * @param[in] size The size in bytes of the memory occupied by the @p data.
+     * @param[in] rpath A resource directory path, if the @p data needs to access any external resources.
+     * @param[in] copy If @c true the data are copied into the engine local buffer, otherwise they are not.
+     *
+     * @retval Result::Success When succeed.
+     * @retval Result::InvalidArguments In case no data are provided or the @p size is zero or less.
+     * @retval Result::NonSupport When trying to load a file with an unsupported extension.
+     * @retval Result::Unknown If an error occurs at a later stage.
+     *
+     * @warning: It's the user responsibility to release the @p data memory.
+     *
+     * @note Experimental API
+     *
+     * @see Text::unload(const std::string& fontId)
+     */
+    static Result load(const std::string& name, const char* data, uint32_t size, const std::string& rpath = "", bool copy = false) noexcept;
```

Comment:
1. `name` is really necessary? I think we can remove it.
2. `rpath` neither.

Reply:
Probably the api doc could address the `name`  should have the `font name` to guide clearly.

Reply:
> > 1. and how will I refer to this specific font after loading it?
> 
> Ah yes, it needs to be. My fault. Let's use `const char* name` instead.

I using string so this will be consistent with unload() api, which expects now path to be a string. if I change name to be const char* probably I should also change path to be  const char*?

Reply:
Ok, let's use a string for now. Just... there is a conflict between the name (string) and font name (char). Let's clean it up under the experimental API later. Probably we should confirm this issue(https://github.com/thorvg/thorvg/issues/2200) and come up with a proper solution.

Reply:
Considering this further, we have this option as well:
```
load("name", data, ...)       //load the name data
load("name", nullptr, ...)   //unload the name data.
```
```
load(path);           //load the path data
unload(path);       //unload the path data
```

Reply:
> say we support both, ttf and woff, and we load two Arial fonts from memory, one from ttf and one from woff file. unloading using name will not work properly, so we should add mimetype to the unload api

In the end, I would like to replace all `std::string` usages with `char*`, both external and internal, to save memory and reduce size if there are no technical issues. `std::string` uses significantly more binary size in comparison.

> so maybe instead of that your approach is better. although it is counterintuitive to use load to unload, but I assume we don't want to add secone api to unload. right? 

I believe users can understand it if we properly address the load API doc. Thus, I don't think it's necessary to add unload() additionally.